src/io/flutter/run/test/FlutterTestRunner.java

```diff
@@ -62,9 +71,64 @@ public boolean canRun(@NotNull String executorId, @NotNull RunProfile profile) {
   @Override
   protected RunContentDescriptor doExecute(@NotNull RunProfileState state, @NotNull ExecutionEnvironment env)
     throws ExecutionException {
-    return runInDebugger((TestLaunchState)state, env);
+    if (env.getExecutor().getId().equals(ToolWindowId.RUN)) {
+      return run((TestLaunchState)state, env);
+    } else {
+      return runInDebugger((TestLaunchState)state, env);
+    }
   }
 
+  protected RunContentDescriptor run(@NotNull TestLaunchState launcher, @NotNull ExecutionEnvironment env)
+    throws ExecutionException {
+    final ExecutionResult executionResult = launcher.execute(env.getExecutor(), this);
+    final ObservatoryConnector connector = new Connector(executionResult.getProcessHandler());
+
+    ApplicationManager.getApplication().executeOnPooledThread(() -> {
+      // Poll, waiting for "flutter run" to give us a websocket.
+      // This is adapted from DartVmServiceDebugProcess::scheduleConnect
+      String url = connector.getWebSocketUrl();
+
+      while (url == null) {
```

Comment:
In `DartVmServiceDebugProcess::scheduleConnect` there's a check for whether the debug session has ended before the connection starts, but I'm not sure what the equivalent is to check for here.

Reply:
Is there a process we can query here, to see if it's still alive? I image there's something in the run state that we can check.

Reply:
The other changes (in the new commit) look good.

I assume from the question here that you'd want to find a way to address this (checking for process termination) before landing this PR?

Reply:
Yeah I didn't find an easy way to do this but figured I'd look around a little more.

Reply:
Exposing the process handler status seems to work here.